packages/core/test/test-util/test-services/v4/test-service/function-imports.ts

```diff
@@ -87,6 +87,46 @@ export function testFunctionImportEntityReturnTypeCollection(parameters: TestFun
   return new FunctionImportRequestBuilderV4('/sap/opu/odata/sap/API_TEST_SRV', 'TestFunctionImportEntityReturnTypeCollection', (data) => transformReturnValueForEntityListV4(data, TestEntity), params);
 }
 
+/**
+ * Type of the parameters to be passed to [[testFunctionImportSharedEntityReturnType]].
+ */
+export interface TestFunctionImportSharedEntityReturnTypeParameters {
+}
+
+/**
+ * Test Function Import Shared Entity Return Type.
+ *
+ * @param parameters - Object containing all parameters for the function import.
+ * @returns A request builder that allows to overwrite some of the values and execute the resultng request.
+ */
+export function testFunctionImportSharedEntityReturnType(parameters: TestFunctionImportSharedEntityReturnTypeParameters): FunctionImportRequestBuilderV4<TestFunctionImportSharedEntityReturnTypeParameters, any> {
+  const params = {
+
+  }
+
+  return new FunctionImportRequestBuilderV4('/sap/opu/odata/sap/API_TEST_SRV', 'TestFunctionImportSharedEntityReturnType', (data) => throwErrorWhenReturnTypeIsUnionType(data, 'TestFunctionImportSharedEntityReturnType'), params);
```

Comment:
Now that I see this, I actually have another idea, that might be better or might be overkill. Maybe the `.execute` method should not exist if this cannot be deserialized. That way we wouldn't have to throw an error, it just wouldn't exist.
I think an even better (and easier) solution might be, that is exists, but it does not exist on the type, so that you get an error at design time and in case you are using js you get a meaningful error at runtime (as it is now). What do you think?

Reply:
Another thought: this actually executes the request before throwing the error. I don't have a good idea at the moment, but if we know that this will happen here already, it doesn't seem appropriate.
The only valid idea that I have right now (and don't like too much) would be to have a `NoExecuteFnImportRB` base class, that either has no `.execute` function or maybe a protected one, that can still be invoked at runtime, but shows you an error at design time.

Reply:
> I think an even better (and easier) solution might be, that is exists, but it does not exist on the type, so that you get an error at design time

Can you explain a bit how this would work if we want to ship pregenerated clients, which contain this ERROR in typescript. Would this not lead to problems?

Reply:
I guess [this](https://www.typescriptlang.org/tsconfig#stripInternal) is the answer.

Reply:
I tested, with this config (annotation in the `.ts` file and enable it in the tsconfig), it can generate a function import, where
- the `.d.ts` does not contain the `execute` function.
- the `.js` contains the `execute` function.

Reply:
Ah, this could be a good options to remember, but than people would still not know why the execute is not there? We can do it like this, but if we get questions like `where is my function import` we should reconsider.

Reply:
btw, there is also a warning about this option
> This is an internal compiler option; use at your own risk

Reply:
Compiler option? Huh? 🤔

Reply:
Still the question remains, should we throw the error before executing?